internal/metadata/git.go

```diff
@@ -60,7 +106,43 @@ func collectGitMetadata(repo *git.Repository) (*GitMetadata, error) {
 
 	headReference := head.Name()
 	if headReference.IsBranch() {
-		metadata.Branch = head.Name().Short()
+		metadata.Branch = headReference.Short()
+	} else {
+		// handle detached HEAD
+		commit, err := repo.CommitObject(head.Hash())
+		if err != nil {
+			return nil, fmt.Errorf("failed to read commit at HEAD")
+		}
+
+		// assume HEAD is a merge commit such as one created by GitHub
+		// this attempts to retrieve a branch name from the last parent and as such,
+		// this doesn't support multi-merge
+		lastParentIndex := commit.NumParents() - 1
+		if lastParentIndex == 1 {
+			parentCommit, err := commit.Parent(lastParentIndex)
+			if err != nil {
+				return nil, fmt.Errorf("failed to resolve parent commit for HEAD")
+			}
+
+			branches, err := repo.Branches()
+			if err != nil {
+				return nil, fmt.Errorf("failed to read branches from repository: %w", err)
+			}
+
+			ref, err := findReference(repo, parentCommit.Hash, branches)
+			if err != nil {
+				return nil, fmt.Errorf("failed to find reference: %w", err)
+			}
+
+			if ref.Name().IsBranch() {
+				metadata.Branch = ref.Name().Short()
+
+				// use the parent reference instead of HEAD because we're unlikely to
+				// find tags on a detached HEAD
+				head = ref
+				fmt.Printf("merge commit in HEAD, assuming branch %s\n", metadata.Branch)
+			}
```

Comment:
I've added a log message to notify the user that we're using a branch from the merge commit.

Reply:
Thanks, looks good!